test/DynamoCoreTests/SerializationTests.cs

```diff
@@ -336,27 +317,207 @@ private void CompareWorkspaces(WorkspaceComparisonData a, WorkspaceComparisonDat
+                                ReferenceLoopHandling = ReferenceLoopHandling.Ignore
+                            });
+
+            var dataPath = filePathBase + ".data";
+            if (File.Exists(dataPath))
             {
-                var vala = kvp.Value;
-                var valb = b.InputsMap[kvp.Key];
-                Assert.AreEqual(vala, valb, "input datas are not the same.");
+                File.Delete(dataPath);
+            }
+            File.WriteAllText(dataPath, dataMapStr);
+        }
+
+        /// <summary>
+        ///  saves workspace comparison object and remaps ids for saved data to the new ids
+        /// </summary>
+        /// <param name="wcd1"></param>
+        /// <param name="filePathBase"></param>
+        /// <param name="executionDuration"></param>
+        /// <param name="modelsGuidToIdMap"></param>
+        public static void SaveWorkspaceComparisonDataWithNonGuidIds(serializationTestUtils.WorkspaceComparisonData wcd1,
+            string filePathBase,
+            TimeSpan executionDuration,
+            Dictionary<Guid,string> modelsGuidToIdMap)
+        {
+            var nodeData = new Dictionary<string, Dictionary<string, object>>();
+            foreach (var d in wcd1.NodeDataMap)
+            {
+                var t = wcd1.NodeTypeMap[d.Key];
+                var nodeDataDict = new Dictionary<string, object>();
+                nodeDataDict.Add("nodeType", t.ToString());
+                nodeDataDict.Add("portValues", d.Value);
+                nodeData.Add(d.Key.ToString(), nodeDataDict);
+            }
+
+            var workspaceDataDict = new Dictionary<string, object>();
+            workspaceDataDict.Add("nodeData", nodeData);
+            workspaceDataDict.Add("executionDuration", executionDuration.TotalSeconds);
+
+            var dataMapStr = JsonConvert.SerializeObject(workspaceDataDict,
+                            new JsonSerializerSettings()
+                            {
+                                Formatting = Formatting.Indented,
+                                ReferenceLoopHandling = ReferenceLoopHandling.Ignore
+                            });
+            //replace all the guids in the data file with all of our remapped ids.
+            foreach (var guidKey in modelsGuidToIdMap.Keys)
+            {
+                dataMapStr = dataMapStr.Replace(guidKey.ToString(), modelsGuidToIdMap[guidKey]);
+            }
+
+            var dataPath = filePathBase + ".data";
+            if (File.Exists(dataPath))
+            {
+                File.Delete(dataPath);
+            }
+            File.WriteAllText(dataPath, dataMapStr);
+        }
+
+        public class PortComparisonData
+        {
+            public string ID { get; set; }
+            public bool UseLevels { get; set; }
+            public bool KeepListStructure { get; set; }
+            public int Level { get; set; }
+            public bool UsingDefaultValue { get; set; }
+
```

Comment:
description is not required here? (AVP has it :) )

Reply:
will look into that.

Reply:
addressed